### PR TITLE
support Arabic

### PR DESCRIPTION
**Pull request by shokai**
Arabic letters followed by a combining mark will form a ligature.

## specs

- Arabic
  - https://www.unicode.org/charts/PDF/U0600.pdf
  - https://unicode-table.com/blocks/arabic/
- Arabic supplement
  - https://www.unicode.org/charts/PDF/U0750.pdf
  - https://unicode-table.com/blocks/arabic-supplement/

## example

نوهُينِ is 5 characters, but 14 bytes.

It can now be split into 5 characters, like this

image

### CODE REVIEW

File: test/arabic.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    const note = 'نِيهُون'
+    assert.lengthOf(note, 7)
+    testBreak(regExp, note, [
+      'نِ',
+      'ي',
+      'هُ',
+      'و',
+      'ن'
+    ])
```
Comment (shokai) [inline test/arabic.js:29]: char length is 7, but split-graphemes breaks 5.

File: src/arabic.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+//  https://www.unicode.org/charts/PDF/U0750.pdf
+//  https://unicode-table.com/blocks/arabic-supplement/
+
+const arabicLetter = '\u0620-\u064A\u066E-\u066F\u0671-\u06D5\u06EE-\u06EF\u06FA-\u06FF'
+const arabicSupplementLetter = '\u0750-\u077F'
+
+const letter = `[${arabicLetter}${arabicSupplementLetter}]`
+const combiningMark = '[\u064B-\u065F\u0670]'
+
+export const arabic = `${letter}${combiningMark}?`
```
Comment (daiiz) [inline src/arabic.js:14]:
```suggestion
export const arabic = `${letter}${combiningMark}*`
```

**Review by daiiz — APPROVED** [review]
LGTM!!

**Comment by shokai** [comment]
Thanks for the review, I'll merge and npm publish.

**Pull request merged**